test/services/task_list_merge_test.dart

```diff
@@ -0,0 +1,189 @@
+    expect(taskLists.first.elements.toSet(), {task1, task2});
+    expect(
+      (await devices[0].taskListService.getTasksFromList('id')).toSet(),
+      {task1, task2},
+    );
+    expect((await devices[0].taskListService.allTasks).toSet(), {task1, task2});
+  });
+
+  test('crdt merge clock drift', () async {
+    // This test is highly dependent on the underlying crdt implementation.
+    // The flutter crdt library throws a ClockDriftException when trying to merge with a Hlc timestamp that is more than 1 minute in the future.
+    // The new implementation should not care.
+
+    final taskList1 = TaskList(id: 'id1', title: 'taskList1');
+    final taskList2 = TaskList(id: 'id2', title: 'taskList2');
+    await devices[0].taskListService.upsertTaskList(taskList1);
+    await devices[1].taskListService.upsertTaskList(taskList2);
+
+    final Map<String, dynamic> device1Crdt =
+        jsonDecode(await devices[1].taskListService.crdtToJson());
+    final Map<String, dynamic> device1Tasks = device1Crdt['records'];
+    expect(device1Tasks.length, 1);
+    final Map<String, dynamic> device1Task1 = device1Tasks.values.first;
+    expect(device1Task1.containsKey('clock'), true);
+    expect(device1Task1['clock'].containsKey('timestamp'), true);
+    device1Task1['clock']['timestamp'] =
+        DateTime.now().add(Duration(hours: 1)).millisecondsSinceEpoch;
+
+    await devices[0].taskListService.mergeCrdtJson(jsonEncode(device1Crdt));
+    expect(
+      (await devices[0].taskListService.taskLists).toSet(),
+      {taskList1, taskList2},
+    );
+  });
+
+  test('crdt update less recent task in list', () async {
+    final taskList = TaskList(id: 'id', title: 'list');
+    final task1 = Task(id: 'task1id', title: 'task1');
+    final task2 = Task(id: task1.id, title: 'task2');
+    await devices[0].taskListService.upsertTaskList(taskList);
+    await devices[0].taskListService.upsertTask(taskList.id!, task1);
+    await Future.delayed(Duration(milliseconds: 10));
+    await devices[1].taskListService.upsertTaskList(taskList);
+    await devices[1].taskListService.upsertTask(taskList.id!, task2);
+
+    await devices[0]
+        .taskListService
+        .mergeCrdtJson(await devices[1].taskListService.crdtToJson());
+    final mergedTaskLists =
+        (await devices[0].taskListService.taskLists).toList();
+    expect(mergedTaskLists.length, 1);
+    final mergedTaskList = mergedTaskLists.first;
+    expect(mergedTaskList.title, taskList.title);
+    expect(mergedTaskList.elements.toSet(), {task2});
+  });
+
+  test('crdt keep more recent task in list', () async {
+    final taskList = TaskList(id: 'id', title: 'list');
+    final task1 = Task(id: 'task1id', title: 'task1');
+    final task2 = Task(id: task1.id, title: 'task2');
+    await devices[1].taskListService.upsertTaskList(taskList);
+    await devices[1].taskListService.upsertTask(taskList.id!, task2);
+    await Future.delayed(Duration(milliseconds: 10));
```

Comment:
Same as above, can we instead set explicit timestamps? Not only does this "slow down" the tests, delays like this may also create non-reproducible test failures.